src/main/java/net/sf/jabref/exporter/BibDatabaseWriter.java

```diff
@@ -208,162 +197,60 @@ public SaveSession savePartOfDatabase(BibDatabaseContext bibDatabaseContext, Lis
                         entryType -> typesToWrite.put(entryType.getName(), entryType));
```

Comment:
With a small difference: the first check is for `StandardType` (i.e. if it's a JabRef default type) and the second one a `KnownType` (i.e. standard or user-specfied).